.github/workflows/update-core.yaml

```diff
@@ -0,0 +1,43 @@
+name: Update Core to Latest Commit
+
+on:
+  schedule:
+    - cron: '0 0 * * *' # every day at midnight
+
+jobs:
+  update-pomerium-core:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b
+        with:
+          fetch-depth: 0
```

Comment:
We don't need the entire repository history for this, do we? I think it should be fine to accept the default shallow clone behavior (see the [README](https://github.com/actions/checkout#:~:text=Only%20a%20single%20commit%20is%20fetched%20by%20default%2C%20for%20the%20ref/SHA%20that%20triggered%20the%20workflow.%20Set%20fetch%2Ddepth%3A%200%20to%20fetch%20all%20history%20for%20all%20branches%20and%20tags.)).